src/stream.ts

```diff
@@ -3108,27 +3108,53 @@ export class Part extends Stream {
         return tempM.estimateStaffLength();
     }
 
-    systemWidthsAndBreaks(): [number[], number[]] {
-        const measureWidths = this.getMeasureWidths();
+    /**
+     * Calculate system breaks and update measure widths as necessary on
+     * account of the reiteration of clefs and key signatures on subsequent systems.
+     */
+    systemWidthsAndBreaks(setMeasureWidths: boolean = true): [number[], number[]] {
         const maxSystemWidth = this.maxSystemWidth; // cryptic note: 'of course fix!'?
-        const systemCurrentWidths = [];
-        const systemBreakIndexes = [];
+        const systemCurrentWidths: number[] = [];
+        const systemBreakIndexes: number[] = [];
         let lastSystemBreak = 0; /* needed to ensure each line has at least one measure */
-        const startLeft = 20; /* TODO: make it obtained elsewhere */
-        let currentLeft = startLeft;
-        for (let i = 0; i < measureWidths.length; i++) {
-            const currentRight = currentLeft + measureWidths[i];
+        // add this padding to each system: it might be needed for a brace
+        const firstMeasurePadding = 20; /* TODO: make it obtained elsewhere */
+
+        let currentSystemIndex = 0;
+        let currentLeft = firstMeasurePadding;
+        for (const [i, m] of Array.from(this.measures).entries()) {
+            const currentRight = currentLeft + m.renderOptions.width;
+            if (i === 0) {
+                m.renderOptions.startNewSystem = true;
+                m.renderOptions.displayClef = true;
+                m.renderOptions.displayKeySignature = true;
+            }
             /* console.log('left: ' + currentLeft + ' ; right: ' + currentRight + ' ; m: ' + i); */
             if (currentRight > maxSystemWidth && lastSystemBreak !== i) {
+                /* first measure of new System */
                 systemBreakIndexes.push(i - 1);
                 systemCurrentWidths.push(currentLeft);
-
-                // console.log('setting new width at ' + currentLeft);
-                currentLeft = startLeft + measureWidths[i]; // 20 + this width;
                 lastSystemBreak = i;
-            } else {
-                currentLeft = currentRight;
+                currentSystemIndex += 1;
+
+                m.renderOptions.displayClef = true;
+                m.renderOptions.displayKeySignature = true;
+                m.renderOptions.startNewSystem = true;
+                m.renderOptions.left = firstMeasurePadding;
+                if (setMeasureWidths) {
+                    m.renderOptions.width = m.estimateStaffLength() + m.renderOptions.staffPadding;
+                }
+            } else if (i !== 0) {
+                m.renderOptions.startNewSystem = false;
+                m.renderOptions.displayClef = false; // check for changed clef first?
+                m.renderOptions.displayKeySignature = false; // check for changed KS first?
+                m.renderOptions.left = currentLeft;
```

Comment:
This is where the renderOptions and some sort of a status marker are in conflict -- we are using "startNewSystem" to mean both "override layout to start new system" and also "in this render should I start a new system?"

Reply:
True, but this was status quo. :-) shall we make an issue or fix here? perhaps these should be set to `undefined`?

Reply:
I think that there should already be an issue for it.  Fixing it here is not on the agenda probably.  But want to keep mentioning it until everyone's sick of hearing it and one of us just goes and fixes it.